command/alloc_logs.go

```diff
@@ -195,77 +208,95 @@ func (l *AllocLogsCommand) Run(args []string) int {
 
 	// If -task isn't provided fallback to reading the task name
 	// from args.
-	if task != "" {
-		err = validateTaskExistsInAllocation(task, alloc)
+	if l.task != "" {
+		err = validateTaskExistsInAllocation(l.task, alloc)
 	} else {
 		if len(args) >= 2 {
-			task = args[1]
-			if task == "" {
+			l.task = args[1]
+			if l.task == "" {
 				l.Ui.Error("Task name required")
 				return 1
 			}
 		} else {
-			task, err = lookupAllocTask(alloc)
+			l.task, err = lookupAllocTask(alloc)
 		}
 	}
 	if err != nil {
 		l.Ui.Error(fmt.Sprintf("Failed to validate task: %s", err))
 		return 1
 	}
 
-	logType := "stdout"
-	if stderr {
-		logType = "stderr"
+	// In order to run the mixed log output, we can only follow the files from
+	// their current positions. There is no way to interleave previous log
+	// lines as there is no timestamp references.
+	if l.follow && !(l.stderr || l.stdout || l.tail || l.numLines > 0 || l.numBytes > 0) {
+		if err := l.tailMultipleFiles(client, alloc); err != nil {
+			l.Ui.Error(fmt.Sprintf("Failed to tail stdout and stderr files: %v", err))
+			return 1
+		}
+	} else {
+
+		// If we are not strictly following the two files, we cannot support
+		// specifying both are targets.
+		if l.stderr && l.stdout {
+			l.Ui.Error("Unable to support both stdout and stderr")
+			return 1
+		}
+
+		logType := api.FSLogNameStdout
+		if l.stderr {
+			logType = api.FSLogNameStderr
+		}
+		if err := l.handleSingleFile(client, alloc, logType); err != nil {
+			l.Ui.Error(fmt.Sprintf("Failed to read %s file: %v", logType, err))
+			return 1
+		}
 	}
 
+	return 0
+}
+
+func (l *AllocLogsCommand) handleSingleFile(client *api.Client, alloc *api.Allocation, logType string) error {
 	// We have a file, output it.
 	var r io.ReadCloser
 	var readErr error
-	if !tail {
-		r, readErr = l.followFile(client, alloc, follow, task, logType, api.OriginStart, 0)
+	if !l.tail {
+		r, readErr = l.followFile(client, alloc, l.follow, l.task, logType, api.OriginStart, 0)
```

Comment:
```suggestion
		r, readErr = l.followFile(client, alloc, logType, api.OriginStart, 0)
```

Since `followFile` is a method and now `follow` and `task` are in `l`, we could remove these, but it's not really necessary 😄